src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs

```diff
@@ -621,6 +609,18 @@ private void CheckStandardTransaction(MempoolValidationContext context, bool wit
                 context.State.Fail(MempoolErrors.Version).Throw();
             }
 
+            if (this.network.Consensus.IsProofOfStake)
+            {
+                long adjustedTime = this.dateTimeProvider.GetAdjustedTimeAsUnixTimestamp();
+                PosFutureDriftRule futureDriftRule = this.consensusRules.GetRule<PosFutureDriftRule>();
+
+                // nTime has different purpose from nLockTime but can be used in similar attacks
+                if (tx.Time > futureDriftRule.GetFutureDrift(adjustedTime))
```

Comment:
Do we have exactly same logic for C++ nodes? 
I mean exactly the same adjusted time calculation, future drift and this rule in place for mempool validation?

Reply:
@dangershony you may have an error here - `GetFutureDrift()` only returns the constant `15`. It does not add the drift amount to the timestamp (the implementation is somewhat different to X, hence the discrepancy). So this would reject every transaction from the mempool I think, as every `.Time` would be `>15`.